credentials/tls/certprovider/pemfile/watcher.go

```diff
@@ -191,23 +198,57 @@ func (w *watcher) updateRootDistributor() {
 		return
 	}
 
+	// Prefer SPIFFEBundleMap, if it works don't use the RootFile
+	err := w.maybeUpdateSPIFFEBundleMap()
+	if err != nil {
+		// The preferred method (spiffe bundle map) failed, try updating the root file
+		_ = w.maybeUpdateRootFile()
+	}
+}
+
+func (w *watcher) maybeUpdateSPIFFEBundleMap() error {
+	// If the map file is unset, just return an error, don't create log spam.
+	if w.opts.SPIFFEBundleMapFile == "" {
+		return fmt.Errorf("SPIFFEBundleMapFile is unset in watcher options")
+	}
+	spiffeBundleMapContents, err := os.ReadFile(w.opts.SPIFFEBundleMapFile)
+	if err != nil {
+		logger.Warningf("spiffeBundleMapFile (%s) read failed: %v", w.opts.SPIFFEBundleMapFile, err)
```

Comment:
Do we really want to log an error here and below? Should whoever calls this function be logging instead?

Reply:
SG, as long as we are not double logging, I'm fine. :)

LGTM.